src/NUnit-Executor/NUnitExecutor.cs

```diff
@@ -1,61 +1,169 @@
+        public IComputer Computer { get; set; }
+        public TestQueue TestQueue { get; set; }
 
         /// <seealso cref="IPlugin.Execute()"/>
         public bool Execute()
         {
-            var test = _testQueue.DequeueTestToRun();
+            Log.WriteInfo("===== Starting tests on Computer: {0}", Computer.Hostname);
+
+            var test = TestQueue.DequeueTestToRun() as NUnitTest;
 
             while (test != null)
             {
                 RunTest(test);
-                test = _testQueue.DequeueTestToRun();
+                test = TestQueue.DequeueTestToRun() as NUnitTest;
             }
 
+            Log.WriteInfo("===== Finished all tests on Computer: {0}", Computer.Hostname);
+
             return true;
         }
 
-        #endregion Inherited from IPlugin
+        /// <seealso cref="IExecutorPlugin.ClonePlugin()"/>
+        public IExecutorPlugin ClonePlugin()
+        {
+            var newPlugin = new NUnitExecutorPlugin(_xmlConfig);
+            newPlugin.Options = Options;
+
+            newPlugin.Name = Name;
+            newPlugin.Computer = Computer;
+            newPlugin.TestQueue = TestQueue;
+
+            return newPlugin;
+        }
+
+        #endregion Inherited from IExecutorPlugin
 
-        private void RunTest(Test test)
+        private void RunTest(NUnitTest test)
         {
             // TODO: Run the test.
+            string baseOutputFile = StringUtils.FormatInvariant(@"{0}\{1}", Options.OutputDirectory, test.TestName);
+
+            string arguments = StringUtils.FormatInvariant(@"{0} /nologo /labels /run:{1} /out:{2}.txt /xml:{2}.xml /timeout:{3}",
+                test.DllPath, test.TestName, baseOutputFile, Computer.Timeout * 1000);
+
+            if (Computer.WorkingDirectory != null)
+            {
+                arguments = StringUtils.FormatInvariant("{0} /work:{1}", arguments, Computer.WorkingDirectory);
+            }
+
+            var processStartInfo = new ProcessStartInfo(NunitPath, arguments);
+            processStartInfo.CreateNoWindow = true;
+            processStartInfo.ErrorDialog = false;
+            processStartInfo.UseShellExecute = false;
+
+            if (Computer.WorkingDirectory != null)
+            {
+                processStartInfo.WorkingDirectory = Computer.WorkingDirectory;
+            }
 
+            foreach (var envKeyValue in Computer.EnvironmentVariables)
+            {
+                Log.WriteDebug("** On Computer '{0}', setting env var '{1}' = '{2}'", Computer.Hostname, envKeyValue.Key, envKeyValue.Value);
+                processStartInfo.EnvironmentVariables.Add(envKeyValue.Key, envKeyValue.Value);
+            }
+
+            Log.WriteInfo("----- Running test: '{0}' on: {1}", test.TestName, Computer.Hostname);
+            Log.WriteDebug("***** Running: {0} {1}", NunitPath, arguments);
+            var process = Process.Start(processStartInfo);
+            bool finished = process.WaitForExit((Computer.Timeout + 60) * 1000);

```

Comment:
Add a comment explaining we're adding an extra 60 seconds to the timeout because NUnit should timeout for us, but if it takes too long to timeout, this will kill it.